app/controllers/impersonate_controller.rb

```diff
@@ -1,6 +1,9 @@
 class ImpersonateController < ApplicationController
   include SecurityHelper
 
+  # This function checks if the logged in user is a student or not. If it is a student, do not allow the impersonate mode.
+  # If the logged in user has the role or anything other than the student, we allow that user to use the impersonate mode.
+
   def action_allowed?
```

Comment:
You can look over the autocomplete guide from [here](https://www.jquerycards.com/forms/autocomplete/rails4-autocomplete/)